lib/mongoose_push_web/plugs/cast_and_validate.ex

```diff
@@ -0,0 +1,133 @@
+defmodule MongoosePushWeb.Plug.CastAndValidate do
+  @behaviour Plug
+
+  @impl Plug
+  def init(opts) do
+    OpenApiSpex.Plug.CastAndValidate.init(opts)
+  end
+
+  @impl Plug
+  def call(conn = %{params: %{"service" => _, "alert" => _}}, opts = %{operation_id: _}) do
+    OpenApiSpex.Plug.CastAndValidate.call(conn, opts)
+  end
+
+  @impl Plug
+  def call(conn = %{params: %{"service" => _, "data" => _}}, opts = %{operation_id: _}) do
+    OpenApiSpex.Plug.CastAndValidate.call(conn, opts)
+  end
+
+  @impl Plug
+  def call(conn = %{params: %{"service" => _}}, opts = %{operation_id: _}) do
+    update_schema_and_do_call(conn, opts)
+  end
+
+  @impl Plug
+  def call(conn = %{params: %{"alert" => _}}, opts = %{operation_id: _}) do
+    update_schema_and_do_call(conn, opts)
+  end
+
+  @impl Plug
+  def call(conn = %{params: %{"data" => _}}, opts = %{operation_id: _}) do
+    update_schema_and_do_call(conn, opts)
+  end
+
+  @impl Plug
+  def call(
+        conn = %{
+          private: %{
+            phoenix_controller: controller,
+            phoenix_action: action,
+            open_api_spex: private_data
+          }
+        },
+        opts
+      ) do
+    operation =
+      case private_data.operation_lookup[{controller, action}] do
+        nil ->
+          operationId = controller.open_api_operation(action).operationId
+          operation = private_data.operation_lookup[operationId]
+
+          operation_lookup =
+            private_data.operation_lookup
+            |> Map.put({controller, action}, operation)
+
+          OpenApiSpex.Plug.Cache.adapter().put(
+            private_data.spec_module,
+            {private_data.spec, operation_lookup}
+          )
+
+          operation
+
+        operation ->
+          operation
+      end
+
+    if operation.operationId do
```

Comment:
Is it possible that `operationId` does not exist in the operation map?
If it does not always exist it will throw an error:
```elixir
iex(1)> m = %{a: 1}
%{a: 1}
iex(2)> m.a
1
iex(3)> m.b
** (KeyError) key :b not found in: %{a: 1}
iex(3)> m[:a]
1
iex(4)> m[:b]
nil
```
since you raze the error yourself, I assume you do not want access operator to do it for. Am I right?
```suggestion
    if operation[:operationId] do
```

Reply:
For my best knowledge it should always exist at this point of the request flow. In case it couldn't, I don't see any need for such a defensive programming anyway. Moreover, this part of code is also copied from the original plug. Please refer to `OpenApiSpex.Plug.CastAndValidate` file content.

Reply:
Since `:operationId` should be always there it does not matter much, but it is not about writing defensive code, it is about throwing error in the right place.

Reply:
Maybe I am repeating myself, but this code originates from openapispex repository and for the reasons I already elaborated on I decided to keep it untouched. Please head to their repo and raise an issue if you think there is any.